src/compiler/transformers/esnext.ts

```diff
@@ -16,9 +19,67 @@ namespace ts {
                 return node;
             }
             switch (node.kind) {
+                case SyntaxKind.BinaryExpression:
+                    const binaryExpression = <BinaryExpression>node;
+                    if (isLogicalOrCoalescingAssignmentOperator(binaryExpression.operatorToken.kind)) {
+                        return transformLogicalAssignmentOperators(binaryExpression);

```

Comment:
nit: Use the singular for "Operator" here (or remove it altogether):
```suggestion
                        return transformLogicalAssignmentOperator(binaryExpression);
```
```suggestion
                        return transformLogicalAssignment(binaryExpression);
```